pkg/didcomm/protocol/didexchange/persistence/connection.go

```diff
@@ -49,6 +64,62 @@ func (c *ConnectionRecorder) SaveInvitation(key string, value interface{}) error
 	return c.store.Put(storeKey, bytes)
 }
 
+// SaveConnectionRecorder saved the connection record against the connection id  in the store
+func (c *ConnectionRecorder) SaveConnectionRecorder(invitation *didexchange.Invitation,
+	msg *didexchange.Message) error {
+	connRecord := &connectionRecord{ThreadID: msg.ThreadID,
+		ServiceEndPoint: invitation.ServiceEndpoint,
+		RecKey:          invitation.RecipientKeys, State: msg.State}
+
+	connBytes, err := json.Marshal(connRecord)
+	if err != nil {
+		return fmt.Errorf("error while marshalling save connection record %s : ", err)
+	}
+	return c.store.Put(msg.ConnectionID, connBytes)
+}
+
+// SaveProtocolID saves the hash of the protocol id which is a did and thread id as an ID and connection id as value.
+func (c *ConnectionRecorder) SaveProtocolID(myDid, thid, connectionID string) error {
+	protocolID, err := computeHash([]byte(myDid + thid))
+	if err != nil {
+		return fmt.Errorf("error while computing hash : %s", err)
+	}
+	return c.store.Put(protocolID, []byte(connectionID))
+}
+
+// FindByProtocolID returns the connection ID based on hashed protocol ID
+func (c *ConnectionRecorder) FindByProtocolID(myDID, thid string) (string, error) {
```

Comment:
Find is not the usual prefix in this project.

Reply:
renamed it to start with get